### PR TITLE
Added status API

### PR DESCRIPTION
**Pull request by atultherajput**
### :chains: Related Resources
https://chefio.atlassian.net/browse/CHEF-2127
https://github.com/chef/automate/pull/7840
### :white_check_mark: Checklist

**All PRs** must tick these:

- [ ] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [ ] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [ ] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [ ] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [ ] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [ ] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable
Screenshot 2023-05-10 at 12 49 03 PM

Screenshot 2023-05-10 at 12 01 19 PM

Screenshot 2023-05-10 at 12 50 12 PM

Screenshot 2023-05-10 at 12 52 40 PM

Screenshot 2023-05-10 at 5 15 07 PM

Demo Video: https://progresssoftware.sharepoint.com/:v:/s/ChefCoreC/EYkV8MvFpThKg_VGL8plml4BNvF3tOdKagtbqlUnDe5sag?e=iRKUfn

### CODE REVIEW

**Review by vivekshankar1 — APPROVED** [review]

File: components/automate-cli/pkg/verifyserver/services/statusservice/statusservice.go [hunk 1]
```diff
@@ -1,19 +1,146 @@
...
+	ExecuteShellCommandFunc func(cmd string) ([]byte, error)
+	Log                     logger.Logger
 }
 
-type StatusService struct{}
+const (
+	AutomateStatusCmd       = "chef-automate status"
+	HabStatusCmd            = "HAB_LICENSE=accept-no-persist hab svc status"
+	AutomateStatusOnBE      = "FileAccessError: Unable to access the file or directory: Failed to read deployment-service TLS certificates: Could not read the service cert: open /hab/svc/deployment-service/data/deployment-service.crt: no such file or directory"
+	automateStatusUnhealthy = "UnhealthyStatusError: System status is unhealthy: One or more services are unhealthy"
```
[the remark this answers is not in the document]
Comment (atultherajput) [reply components/automate-cli/pkg/verifyserver/services/statusservice/statusservice.go]: done

**Review by punitmundra — APPROVED** [review]
looks fine.
just replace all string literal with constant 
and update the comment over Parse API ( in case of hab svc status output change in future need modification in Parsing function).

File: components/automate-cli/pkg/verifyserver/utils/fiberutils/os.go [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package fiberutils
```
Comment (Dmaddu) [inline components/automate-cli/pkg/verifyserver/utils/fiberutils/os.go:1]: missing test file

File: components/automate-cli/pkg/verifyserver/services/statusservice/statusservice.go [hunk 3]
```diff
@@ -1,19 +1,141 @@
...
-	GetServices() []models.ServiceDetails
+	GetServices() (*[]models.ServiceDetails, error)
+}
+
+type StatusService struct {
+	ExecuteShellCommandFunc func(cmd string) ([]byte, error)
+	Log                     logger.Logger
+}
+
+func NewStatusService(executeShellCommand func(cmd string) ([]byte, error), log logger.Logger) IStatusService {
```
Comment (Dmaddu) [inline components/automate-cli/pkg/verifyserver/services/statusservice/statusservice.go:22]: NewStatusService should return object of StatusService not interface

**Review by Dmaddu — CHANGES_REQUESTED** [review]

**Review by vviveksharma — APPROVED** [review]

**Review by prasad927 — APPROVED** [review]

File: components/automate-cli/pkg/verifyserver/utils/fiberutils/os.go (hunk 2 above)
Comment (atultherajput) [reply components/automate-cli/pkg/verifyserver/utils/fiberutils/os.go:1]: done

File: components/automate-cli/pkg/verifyserver/services/statusservice/statusservice.go (hunk 3 above)
Comment (atultherajput) [reply components/automate-cli/pkg/verifyserver/services/statusservice/statusservice.go]: done

**Review by Dmaddu — APPROVED** [review]

**Pull request merged**